sim/battle.ts

```diff
@@ -3317,148 +3246,94 @@ export class Battle extends Dex.ModdedDex {
 		return this.sides[parseInt(sideid[1], 10) - 1];
 	}
 
-	runMove(
-		move: string | Move, target: Pokemon, targetLoc?: number,
-		sourceEffect?: Effect | null, zMove?: string, externalMove?: boolean
-	) {
-		throw new Error(`The runMove function needs to be implemented in scripts.js or the battle format.`);
-	}
-
-	useMove(
-		move: string | Move, pokemon: Pokemon, target?: Pokemon | null,
-		sourceEffect?: Effect | null, zMove?: string
-	): boolean {
-		throw new Error(`The useMove function needs to be implemented in scripts.js or the battle format.`);
+	afterMoveSecondaryEvent(targets: Pokemon[], pokemon: Pokemon, move: ActiveMove): undefined {
+		throw newUnimplementedError('afterMoveSecondary');
 	}
 
-	/**
-	 * target = undefined: automatically resolve target
-	 * target = null: no target (move will fail)
-	 */
-	useMoveInner(
-		move: string | Move, pokemon: Pokemon, target?: Pokemon | null,
-		sourceEffect?: Effect | null, zMove?: string
-	): boolean {
-		throw new Error(`The useMoveInner function needs to be implemented in scripts.js or the battle format.`);
+	calcRecoilDamage(damageDealt: number, move: Move): number {
+		throw newUnimplementedError('calcRecoilDamage');
```

Comment:
Why `newUnimplementedError` instead of `new UnimplementedError`?

Reply:
Because from my research properly implementing an Error class seemed kind of obnoxious in JS, but I guess I'm wrong.

Reply:
See e.g. https://github.com/Zarel/Pokemon-Showdown/blob/master/server/loginserver.js#L26-L27

(A few years ago it was really complicated, yes. It even required usage of non-standard JS provided by the V8 engine).

Reply:
Cool. The pattern of setting `this.name` in the constructor also seems to work just the same.